packages/plugin-color/src/Color.ts

```diff
@@ -54,24 +54,16 @@ export class Color<T extends ColorConfig = ColorConfig> extends JWPlugin<T> {
             const currentCache = inline.cache.style || {};
             inline.cache.style = new CssStyle({ ...currentCache, [this.styleName]: color });
         } else {
-            for (const node of params.context.range.selectedNodes()) {
-                // Skip if the node already has the right color, through an
-                // ancestor or a format.
-                if (node.is(InlineNode)) {
-                    const colorFormat = node.modifiers
-                        .filter(Format)
-                        .find(format =>
-                            format.modifiers.find(Attributes)?.style?.has(this.styleName),
-                        );
-                    if (
-                        colorFormat &&
-                        colorFormat.modifiers.get(Attributes).style.get(this.styleName) === color
-                    )
-                        continue;
-                }
+            const selectedNodes = params.context.range.selectedNodes();
+            // Skip if the node already has the right color, through an
+            // ancestor or a format.
+            const notColoredYet = selectedNodes.filter(node => {
+                if (this.hasColor(color, node)) return false;
                 const colorAncestor = node.ancestor(this.hasColor.bind(this));
-                if (colorAncestor && this.hasColor(color, colorAncestor)) continue;
-
+                if (colorAncestor && this.hasColor(color, colorAncestor)) return false;
+                return true;
```

Comment:
What about this ?
```typescript
return !colorAncestor || !this.hasColor(color, colorAncestor);
```

Reply:
It will crash if `colorAncestor` isn't defined.

Reply:
Wait, you can do `if(colorAncestor)` if it is not defined without crashing but you can't do `if (!colorAncestor)` ?